tests/public-api/examples/optimization/solver.py

```diff
@@ -0,0 +1,194 @@
+import os
+import time
+from pathlib import Path
+from tqdm import tqdm
+import shutil
+import json
+import logging
+import zipfile
+from tempfile import TemporaryDirectory
+from typing import Optional, List
+
+import osparc
+from osparc.models import File, Solver, Job, JobStatus, JobInputs, JobOutputs
+from osparc.api import FilesApi, SolversApi
+
+
+logging.basicConfig(level=logging.ERROR, format='[%(levelname)s] %(message)s')
+
+class OsparcSolver():
+    """
+    An oSparc solver
+    """
+    def __init__(self, solver_key: str, solver_version: str, cfg: osparc.Configuration):
+
+        self._solver_key: str = solver_key
+        self._solver_version: str = solver_version
+        self._cfg: osparc.Configuration = cfg
+
+        # APIs
+        try:
+            self._api_client = osparc.ApiClient(cfg)
+            self._users_api = osparc.UsersApi(self._api_client)
+            self._files_api = FilesApi(self._api_client)
+            self._solvers_api = SolversApi(self._api_client)
+            self._users_api.get_my_profile() # validate access
+        except osparc.exceptions.ApiException as e:
+            self._api_client.close()
```

Comment:
TIP: can perhaps create an class-wide deallocation mechanism using a decorator and https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack

This will avoid adding this type of code on every call!

Reply:
Good tip! I have added a decorator to handle any `osparc.exceptions.ApiException` raised in a method in this class. I am not sure exactly what you mean by the "deallocation mechanism".